frontend/src/components/MainNavigation.vue

```diff
@@ -289,26 +289,30 @@ export default {
       }
     },
     projectFilterHasExactMatch () {
-      return this.isProjectNameMatchingFilter(get(head(this.sortedAndFilteredProjectList), 'metadata.name'))
+      const projectName = get(head(this.sortedAndFilteredProjectList), 'metadata.name')
+      return this.isProjectNameMatchingFilter(projectName)
     },
-    selectedProject () {
-      return this.sortedAndFilteredProjectListWithAllProjects[this.selectedProjectIndex]
+    highlightedProject () {
+      return this.sortedAndFilteredProjectListWithAllProjects[this.highlightedProjectIndex]
     }
   },
   methods: {
     ...mapActions([
       'setSidebar'
     ]),
     onEnterProjectFilter () {
-      this.projectMenu = false
-      this.project = this.selectedProject
+      this.navigateToProject(this.highlightedProject)
     },
     onProjectClick (event, project) {
-      this.projectMenu = false
       if (event.isTrusted) {
+        // skip untrusted events - e.g. events triggered via enter key
```

Comment:
... as they are handled via `onEnterProjectFilter`, right? 
if yes, the comment should mention that

Reply:
yes, we listen for the enter key event and navigate to the `highlighted` project. The events we skip here are caused by default browser behavior - the project list items are rendered as links and can be selected using the arrow keys, pressing enter will trigger a mouse-like event (untrusted) for the `focused` link. This is what we skip here as we handle enter events as you mentioned. Moreover, the default selection behavior is behaving incorrectly for our use case